src/main.ts

```diff
@@ -84,32 +84,50 @@ async function getChangedFiles(token: string, base: string, ref: string, initial
     return await git.getChangesOnHead()
   }
 
-  const prEvents = ['pull_request', 'pull_request_review', 'pull_request_review_comment', 'pull_request_target']
-  if (prEvents.includes(github.context.eventName)) {
-    if (ref) {
-      core.warning(`'ref' input parameter is ignored when 'base' is set to HEAD`)
-    }
-    if (base) {
-      core.warning(`'base' input parameter is ignored when action is triggered by pull request event`)
-    }
-    const pr = github.context.payload.pull_request as PullRequestEvent
-    if (token) {
-      return await getChangedFilesFromApi(token, pr)
+  switch (github.context.eventName) {
+    // To keep backward compatibility, commits in GitHub pull request event
+    // take precedence over manual inputs.
+    case 'pull_request':
+    case 'pull_request_review':
+    case 'pull_request_review_comment':
+    case 'pull_request_target': {
+      if (ref) {
+        core.warning(`'ref' input parameter is ignored when 'base' is set to HEAD`)
+      }
+      if (base) {
+        core.warning(`'base' input parameter is ignored when action is triggered by pull request event`)
+      }
+      const pr = github.context.payload.pull_request as PullRequestEvent
```

Comment:
I guess it may not be working currently...?

```suggestion
      const pr = github.context.payload as PullRequestEvent
```